pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -353,3 +376,19 @@ func (w *worker) findNodeTemplateAndMachineTypeByPoolName(ctx context.Context, o
 	}
 	return nil, ""
 }
+
+// CheckWorkerStatusMachineDeployments checks if the status of the worker is updated or not during its reconciliation.
+// It is updated if
+// * The status.MachineDeploymentsLastUpdateTime > the value of the time stamp stored in worker struct before the reconciliation begins.
+func (w *worker) checkWorkerStatusMachineDeployments(o client.Object) error {
+	obj, ok := o.(*extensionsv1alpha1.Worker)
+	if !ok {
+		return fmt.Errorf("expected *extensionsv1alpha1.Worker but got %T", o)
+	}
+
+	if obj.Status.MachineDeploymentsLastUpdateTime != nil && (w.machineDeploymentLastUpdateTime == nil || obj.Status.MachineDeploymentsLastUpdateTime.After(w.machineDeploymentLastUpdateTime.Time)) {
```

Comment:
While I was having a final look I noticed that this check can be problematic for extensions that have not been revendored to the updated logic in `extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go`. In this case `obj.Status.MachineDeploymentsLastUpdateTime` will be `nil` and `checkWorkerStatusMachineDeployments` will always fail.
Can you double check please?

Reply:
Yes it is true that if the extension is not revendored with these changes, then it will always fail. This is because the `status.machineDeploymentsLastUpdateTime` is updated by the controller which extension will run and if extension is not updated then it won't have the code to update the `machineDeploymentsLastUpdateTime`. Sorry, I did not think of this case.

I don't see how `status.machineDeploymentsLastUpdateTime` can be updated by anyone other than the extensions.

Reply:
Yes, that's probably not possible. We need a fallback mechanism for a certain period of time if `status.machineDeploymentsLastUpdateTime` is not maintained, i.e. CA is deployed the way it was before this change.

Reply:
addressed in [07b5f45](https://github.com/gardener/gardener/pull/7774/commits/07b5f452da5803daab51baa811af17e18abf95ca)